NXDOC-2516: Release and upgrade notes for 2021.30 (#2034)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-server/nuxeo-server-release-notes/nuxeo-server-release-notes-2021-30.md

---
title: LTS 2021.30 / LTS 2021-HF30
description: Discover what's new in LTS 2021.30 / LTS 2021-HF30
review:
   comment: ''
   date: '2022-12-07'
   status: ok
labels:
    - release-notes
toc: true
tree_item_index: 0
hidden: true
---

{{! multiexcerpt name='nuxeo-server-updates-2021-30'}}
# What's New in LTS 2021.30 / LTS 2021-HF30

> ## Use jackson bom for jackson dependencies

Reviewer: ```suggestion
## Use Jackson BOM for Jackson Dependencies
```

Jackson BOM is now imported in the Maven dependency graph.

> Addons declaring jackson artefacts in their dependencies managment section can remove them as the `jackson-bom` is now imported in the Maven dependency graph of Nuxeo Platform.

Reviewer: ```suggestion
Addons declaring jackson artefacts in their dependencies management section can remove them as the `jackson-bom` is now imported in the Maven dependency graph of Nuxeo Platform.
```

This will ease the security upgrade of Jackson artefacts for all Nuxeo artefacts.

<i class="fa fa-long-arrow-right" aria-hidden="true"></i>&nbsp;More on JIRA ticket [NXP-31455](https://jira.nuxeo.com/browse/NXP-31455)

> ## Make certificateKeyAlias value configurable from nuxeo.conf 

Reviewer: ```suggestion
## Make certificateKeyAlias Value Configurable from nuxeo.conf 
```

The certification key alias can be configured.

<i class="fa fa-long-arrow-right" aria-hidden="true"></i>&nbsp;More on JIRA ticket [NXP-31425](https://jira.nuxeo.com/browse/NXP-31425)

> ## Upgrade MongoDB Driver to version 4.7

Reviewer: ```suggestion
## Upgrade MongoDB Driver to Version 4.7
```

> The MongoDB driver has been bumped to version 4.7 to allow using MongoDB 6.x 

Reviewer: ```suggestion
The MongoDB driver has been bumped to version 4.7 to allow using MongoDB 6.x.
```

<i class="fa fa-long-arrow-right" aria-hidden="true"></i>&nbsp;More on JIRA ticket [NXP-31398](https://jira.nuxeo.com/browse/NXP-31398)

> ## Make the document creation take into account the segment limit in WebUI

Reviewer: ```suggestion
## Make the Document Creation Take Into Account the Segment Limit in Web UI
```

> Apply path segment max size at document creation

Reviewer: ```suggestion
Apply path segment max size at document creation.
```

> The segment limit can be adapted like in the sample contrib attached

Reviewer: ```suggestion
The segment limit can be adapted like in the sample contrib attached.
```

<i class="fa fa-long-arrow-right" aria-hidden="true"></i>&nbsp;More on JIRA ticket [NXP-31363](https://jira.nuxeo.com/browse/NXP-31363)

> ## Create a MongoTransientStore that can handle large number params

Reviewer: ```suggestion
## Create a MongoTransientStore That Can Handle Large Number Params
```

> There is a new optimized implementation of TransientStore for MongoDB

Reviewer: ```suggestion
There is a new optimized implementation of TransientStore for MongoDB.
```

<i class="fa fa-long-arrow-right" aria-hidden="true"></i>&nbsp;More on JIRA ticket [NXP-31311](https://jira.nuxeo.com/browse/NXP-31311)

> ## Fix race condition when reading Work state from KV store

Reviewer: ```suggestion
## Fix Race Condition When Reading Work State From KV Store
```
